File: bddframework/pkg/framework/client/kubernetes/event.go
// Copyright 2020 Red Hat, Inc. and/or its affiliates
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package kubernetes

import (
	"context"
	v1beta1 "k8s.io/api/events/v1beta1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/kiegroup/kogito-cloud-operator/core/client"
)

// EventInterface has functions that interacts with pod object in the Kubernetes cluster
type EventInterface interface {
	// Retrieve all events from a namespace
	GetEvents(namespace string) (*v1beta1.EventList, error)
}

type event struct {
	client *client.Client
}

func newEvent(c *client.Client) EventInterface {
	return &event{
		client: c,
	}
}

func (event *event) GetEvents(namespace string) (*v1beta1.EventList, error) {
	opts := metav1.ListOptions{}
	return event.client.KubernetesExtensionCli.EventsV1beta1().Events(namespace).List(context.TODO(), opts)
}

File: packages/j2cl-tools/jakarta-spec/enterprise/src/main/java/jakarta/enterprise/event/Event.java
/*
 * JBoss, Home of Professional Open Source
 * Copyright 2010, 2015, Red Hat, Inc., and individual contributors
 * by the @authors tag. See the copyright.txt in the distribution for a
 * full listing of individual contributors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jakarta.enterprise.event;

import jakarta.enterprise.inject.Any;

/**
 * Allows the application to fire events of a particular type.
 *
 * <p>Beans fire events via an instance of the <code>Event</code> interface, which may be injected:
 *
 * <pre>
 * &#064;Inject
 * &#064;Any
 * Event&lt;LoggedInEvent&gt; loggedInEvent;
 * </pre>
 *
 * <p>The <code>fire()</code> method accepts an event object:
 *
 * <pre>
 * public void login() {
 *    ...
 *    loggedInEvent.fire( new LoggedInEvent(user) );
 * }
 * </pre>
 *
 * <p>Any combination of qualifiers may be specified at the injection point:
 *
 * <pre>
 * &#064;Inject
 * &#064;Admin
 * Event&lt;LoggedInEvent&gt; adminLoggedInEvent;
 * </pre>
 *
 * <p>Or, the {@link Any &#064;Any} qualifier may be used, allowing the application to specify
 * qualifiers dynamically:
 *
 * <pre>
 * &#064;Inject
 * &#064;Any
 * Event&lt;LoggedInEvent&gt; loggedInEvent;
 * </pre>
 *
 * <p>For an injected <code>Event</code>:
 *
 * <ul>
 *   <li>the <em>specified type</em> is the type parameter specified at the injection point, and
 *   <li>the <em>specified qualifiers</em> are the qualifiers specified at the injection point.
 * </ul>
 *
 * <p>Events may also be fired asynchronously with {@link #fireAsync(Object)} and {@link
 * #fireAsync(Object, NotificationOptions)} methods
 *
 * @author Gavin King
 * @author Pete Muir
 * @author David Allen
 * @author Antoine Sabot-Durand
 * @param <T> the type of the event object
 */
public interface Event<T> {

  /**
   * Fires an event with the specified qualifiers and notifies observers.
   *
   * @param event the event object
   * @throws IllegalArgumentException if the runtime type of the event object contains a type
   *     variable
   * @throws ObserverException if a notified observer throws a checked exception, it will be wrapped
   *     and rethrown as an (unchecked) {@link ObserverException}
   */
  public void fire(T event);
}
